components/index.css

```diff
@@ -37,6 +37,14 @@
 
 .nav-right a{
     text-decoration: none;
+    padding: 0.5rem 0;
+    color: var(--primary-color);
+}
+
+.nav-right a:hover{
+    transition: all 0.2s ease;
+    border-bottom: 3px solid var(--primary-color);
+    /* background-color:rgb(119, 186, 253);   */
```

Comment:
Add your transition in the main class(.nav-right a) and not on your hover state. Because as soon you remove the hover state no transition will apply.